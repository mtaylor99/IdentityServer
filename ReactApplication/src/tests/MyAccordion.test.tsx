import {describe, expect, test} from 'vitest';
import {render, screen, waitFor} from '@testing-library/react';
import MyAccordion from './MyAccordion';

describe("Accordion test", () => {
    test("should show title all the time", () => {
        
        render(<MyAccordion title='Testing'><h4>Content</h4></MyAccordion>);

        expect(screen.getByText(/Testing/i)).toBeDefined()
    })

    test("Snapshot - Accordion component", async () => {
        const { container, getByText } = render(<MyAccordion title='Testing'><h4>Content</h4></MyAccordion>);
    
        await waitFor(() => getByText(/Testing/i));
    
        expect(container).toMatchSnapshot();
      });

})