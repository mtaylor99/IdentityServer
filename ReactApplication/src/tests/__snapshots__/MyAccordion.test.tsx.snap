// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Accordion test > Snapshot - Accordion component 1`] = `
<div>
  <div
    class="MuiBox-root css-qths1"
  >
    <h3
      class="accordion-title"
    >
      Testing
    </h3>
    <div
      class="accordion-content"
    >
      <h4>
        Content
      </h4>
    </div>
  </div>
</div>
`;
